source-academy/cadet
# Add contest voting

Backend changes to support submission of contest voting "assessments" in the frontend. 

Summary of changes:
Contest Voting: 
* Inclusion of new database table named 'Submission_Votes' used to store Voting entries. 
* Introduction of new 'voting' question type 
* New assessment type 'contest_voting' which is a 'Practical'/others tab in the frontend 
* Operations to support submission/updating of results to store results in database

Leaderboard: 
* Computation of rolling leaderboard scores.
* Serving the frontend to present leaderboard results to users. 



How to Test: 

Checklist: 

Tracking WIP: 
- add relative_score to answers table [DONE] 
- implement cron job for computing vote scores [done] 
- Autograder [done] 
- Fix random assignment function [done]
- Check correctness of compute_scores function implementation detail [done]  
- implement different scoring types on xml_parser (vote only / vote + lex token count)  
Solution: For now, if content to vote for is a youtube link - wrap it in a comment eg. //www.youtube... /* www. */ to prevent the lexer from counting the tokens and deducting penalty
